src/sentry/integrations/slack/sdk_client.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from functools import wraps
+from types import FunctionType
+
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+from slack_sdk.web import SlackResponse
+
+from sentry.models.integrations import Integration
+from sentry.services.hybrid_cloud.integration import integration_service
+from sentry.silo.base import SiloMode
+from sentry.utils import metrics
+
+SLACK_DATADOG_METRIC = "integrations.slack.http_response"
+
+logger = logging.getLogger(__name__)
+
+
+def track_response_data(response: SlackResponse, error: str | None = None) -> None:
+    is_ok = response.get("ok", False)
+    code = response.status_code
+
+    metrics.incr(
+        SLACK_DATADOG_METRIC,
+        sample_rate=1.0,
+        tags={"ok": is_ok, "status": code},
+    )
+
+    extra = {
+        "integration": "slack",
+        "status_string": str(code),
+        "error": error,
+    }
+    logger.info("integration.http_response", extra=extra)
```

Comment:
are we going to know which method failed though? Like which API call? Any chance we can capture that as well to help us understand which method is failing?

Reply:
i think i can pass this in the logger if i pass the method in

Reply:
like method.__name__

Reply:
let me do this in a followup, and move the metaclass stuff to integrations.base. currently the only API call is chat.postMessage